app/src/main/java/com/github/se/icebreakrr/utils/NetworkUtils.kt

```diff
@@ -0,0 +1,39 @@
+package com.github.se.icebreakrr.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.widget.Toast
+
+// Written with the help of CursorAI and Claude
+
+/**
+ * A utility object for network-related functions.
+ *
+ * This object provides methods to check network availability and to show a toast message when there
+ * is no internet connection.
+ */
+object NetworkUtils {
+
+  /**
+   * Checks if the network is available.
+   *
+   * @param context The context used to access system services.
+   * @return True if the network is available, false otherwise.
+   */
+  fun isNetworkAvailable(context: Context): Boolean {
+    val connectivityManager = context.getSystemService(ConnectivityManager::class.java)
```

Comment:
Important: Make this null-safe, as some devices may return null for this system service in unusual conditions